aws-sns-topicpolicy/src/main/java/software/amazon/sns/topicpolicy/CreateHandler.java

```diff
@@ -1,77 +1,71 @@
-                .then(progress -> ProgressEvent.success(model, callbackContext));
-    }
 
-    /**
-     * Invocation of initCallbackContextAndPrimaryIdentifier generates primary identifier, initializes the context &
-     * stores topics in it.
-     *
-     * @param request
-     *            {@link ResourceHandlerRequest}
-     * @param callbackContext
-     *            {@link CallbackContext}
-     * @return {@link ProgressEvent}
-     */
+        logger.log(String.format("[StackId: %s, ClientRequestToken: %s] Calling Create SNS TopicPolicy", request.getStackId(), request.getClientRequestToken()));
+
+        return ProgressEvent.progress(resourceModel, callbackContext)
+                .then(progress -> Create(proxy, proxyClient, request, progress, logger))
+                .then(progress -> ProgressEvent.defaultSuccessHandler(null));
+    }
 
-    private ProgressEvent<ResourceModel, CallbackContext> initCallbackContextAndPrimaryIdentifier(
+    protected ProgressEvent<ResourceModel, CallbackContext> Create(
             final AmazonWebServicesClientProxy proxy,
             final ProxyClient<SnsClient> proxyClient,
             final ResourceHandlerRequest<ResourceModel> request,
-            final CallbackContext callbackContext,
-            ProgressEvent<ResourceModel, CallbackContext> progress) {
-
-        final CallbackContext currentContext = callbackContext == null ? CallbackContext
-                .builder()
-                .build() : callbackContext;
+            ProgressEvent<ResourceModel, CallbackContext> progress,
+            final Logger logger) {
         final ResourceModel model = request.getDesiredResourceState();
-        // setting up primary id if not provided
-        if (StringUtils.isNullOrEmpty(model.getId())) {
-            final String Id = IdentifierUtils.generateResourceIdentifier(
-                    request.getLogicalResourceIdentifier() == null ? "SnsTopicPolicy"
-                            : request.getLogicalResourceIdentifier(),
-                    request.getClientRequestToken(),
-                    MAX_LENGTH_SNS_TOPICPOLICY_ID);
-            model.setId(Id.toLowerCase());
+        final CallbackContext callbackContext = progress.getCallbackContext();
+        final String policy = getPolicyDocument(request);
+        List<String> topics = model.getTopics();
+        for (final String topicArn : topics) {
+            final ProgressEvent<ResourceModel, CallbackContext> progressEvent = proxy
+                    .initiate("AWS-SNS-TopicPolicy::Create", proxyClient, model, callbackContext)
+                    .translateToServiceRequest((resourceModel) -> Translator.translateToRequest(topicArn, policy))
+                    .makeServiceCall((awsRequest, client) -> {
+                        SetTopicAttributesResponse response = proxyClient.injectCredentialsAndInvokeV2(awsRequest, client.client()::setTopicAttributes);
```

Comment:
need to implement this try catch
https://code.amazon.com/packages/AWS21Workflows/blobs/mainline/--/src/com/amazon/aws21/workflows/resources/sns/SnsActivitiesImpl.java#:~:text=try%20%7B,throw%20e%3B

Reply:
need to throw something similar to LimitedRetriableException(e) not CfnInvalidRequestException(e)